internal/install/discovery/mock_discoverer.go

```diff
@@ -24,6 +25,14 @@ func (d *MockDiscoverer) Os(os string) {
 	d.DiscoveryManifest.OS = os
 }
 
+func (d *MockDiscoverer) PlatformVersion(pf string) {
```

Comment:
Should we name this `SetPlatformVersion`?

Reply:
I don't know, the other method "Os" didn't have a "SetOs", so I kept consistency. I kind of like for this type of test object to behave like a builder, and it makes it a bit more readable without the Set?

Reply:
I actually had the same thought as I came across this and paused wondering if it should be `setPaltformVersion`. I'm not sure what's idiomatic w.r.t. Go when it comes to setters.

Reply:
Meh, not gonna nitpick about it, but in my opinion a function call of `PlatformVersion()` semantically feels like it would return me the current platform version and not set it. 

i.e. `os.PlatformVersion()` vs `os.SetPlatformVersion()` have semantically different meanings to me

Either way, not gonna make it a blocker for the PR.

Reply:
I ve updated to a SetXxx semantic